### ISSUE TITLE
Alternate 'migrations' directory

### ISSUE THREAD

**Issue by drusellers** [weavejester/ragtime#58]
I am curious to know if you would be for or against having an additional (optional) field in the ragtime map to change the default directory that ragtime looks in for migrations?

I looked through the code and see that its currently set as a private symbol here https://github.com/weavejester/ragtime/blob/master/ragtime.sql.files/src/ragtime/sql/files.clj#L110

After digging through the code, it looks like the best long term option would be to have an option field that holds data to be passed to the migration method so that other migrators could use more than just a string. So i would do that.

**Comment by kernius** [comment]
+1

We really need a way to specify a different migrations directory.

Related: https://github.com/weavejester/ragtime/pull/29

**Comment by martintrojer** [comment]
In the meantime you can have a look at Joplin https://github.com/juxt/joplin (built on top of ragtime).

**Comment by ThomasMeier** [comment]
A workaround, if you're using the plugin, would be to define a function that runs returns the migrations sequence with the directory supplied.

In `project.clj`

```clj
(defproject sandbox "0.1.0-SNAPSHOT"
  :dependencies [[org.clojure/clojure "1.6.0"]
                 [org.postgresql/postgresql "9.3-1102-jdbc41"]
		         [ragtime "0.3.8"]]
  :plugins [[ragtime/ragtime.lein "0.3.8"]]
  :ragtime {:migrations sandbox.core/custom-migrate-fn
                 :database "jdbc:postgresql://localhost:5432/test_ragtime?user=admin"})
```

And in `core.clj`

```clj
(ns sandbox.core
  (:require [ragtime.sql.files :refer (migrations)])
  (:gen-class))

;; Looks inside src/sql/migrations directory for sql files.
(defn custom-migrate-fn [] 
  (migrations "src/sql/migrations"))
```

**Comment by weavejester** [comment]
Fixed by version 0.4.0 redesign.
